### ISSUE TITLE
`sf project deploy start` behaves incorrectly in Windows Powershell when deploying zip files

### ISSUE THREAD

**Issue by tarcang** [forcedotcom/cli#2581]
### Summary
`sf project deploy start` is not able to deploy single package as a zip file in Powershell. If I run the same command in MacOS bash terminal, the behavior is fine.

### Steps To Reproduce

Prereq: Have an Windows VM or an environment that runs Powershell 5. 

1. Create a scratch org with Commerce Enabled (you can use `sf commerce scratchorg create` given that you have a devhub, from [@salesforce/commerce](https://github.com/forcedotcom/commerce-on-lightning)
2. Download [this test zip file I was using](https://drive.google.com/file/d/1S_rGMxMmt4C5XajcGaNSXRPmZcZRuZVY/view?usp=sharing)
3. Run 
```
sf project deploy start -o <your_scratch_org> -g --metadata-dir='./b2clocal21ToDeploy.zip' -w 60 --verbose --api-version=59.0 --json --single-package
```
4. You will get an output, but not for the individual components. You will get that only package.xml is deployed with no report on the actual components:
```
{
  "status": 0,
  "result": {
    "checkOnly": false,
    "completedDate": "2023-11-28T15:45:14.000Z",
    "createdBy": "005TC000000fG41",
    "createdByName": "User User",
    "createdDate": "2023-11-28T15:45:14.000Z",
    "details": {
      "componentSuccesses": [
        {
          "changed": true,
          "componentType": "",
          "created": false,
          "createdDate": "2023-11-28T15:45:14.000Z",
          "deleted": false,
          "fileName": "package.xml",
          "fullName": "package.xml",
          "success": true
        }
      ],
      "runTestResult": {
        "numFailures": 0,
        "numTestsRun": 0,
        "totalTime": 0,
        "codeCoverage": [],
        "codeCoverageWarnings": [],
        "failures": [],
        "flowCoverage": [],
        "flowCoverageWarnings": [],
        "successes": []
      },
      "componentFailures": []
    },
    "done": true,
    "id": "0AfTC000002ODPi0AO",
    "ignoreWarnings": true,
    "lastModifiedDate": "2023-11-28T15:45:14.000Z",
    "numberComponentErrors": 0,
    "numberComponentsDeployed": 0,
    "numberComponentsTotal": 0,
    "numberTestErrors": 0,
    "numberTestsCompleted": 0,
    "numberTestsTotal": 0,
    "rollbackOnError": true,
    "runTestsEnabled": false,
    "startDate": "2023-11-28T15:45:14.000Z",
    "status": "Succeeded",
    "success": true,
    "files": []
  },
  "warnings": []
}
```

### Expected result
I would get feedback about the components and they should end in success just like in my MacOS system.

### Actual result
Nothing is actually deployed in my org.

### System Information
```json
{
  "architecture": "win32-x64",
  "cliVersion": "@salesforce/cli/2.17.14",
  "nodeVersion": "node-v20.10.0",
  "osVersion": "Windows_NT 10.0.19045",
  "rootPath": "C:\\Users\\tarcan.gul\\AppData\\Roaming\\npm\\node_modules\\@salesforce\\cli",
  "shell": "cmd.exe",
  "pluginVersions": [
    "@mshanemc/plugin-streaming 1.1.7 (user)",
    "@mshanemc/sfdx-sosl 1.1.0 (user)",
    "@oclif/plugin-autocomplete 3.0.1 (core)",
    "@oclif/plugin-commands 3.0.6 (core)",
    "@oclif/plugin-help 6.0.5 (core)",
    "@oclif/plugin-not-found 3.0.3 (core)",
    "@oclif/plugin-plugins 4.1.7 (core)",
    "@oclif/plugin-search 1.0.6 (core)",
    "@oclif/plugin-update 4.1.3 (core)",
    "@oclif/plugin-version 2.0.6 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.2 (core)",
    "@oclif/plugin-which 3.0.8 (core)",
    "@salesforce/cli 2.17.14 (core)",
    "@salesforce/commerce 248.1.3 (user)",
    "apex 2.3.20 (core)",
    "auth 2.8.26 (core)",
    "community 2.4.8 (user)",
    "data 2.6.3 (core)",
    "deploy-retrieve 1.20.0 (core)",
    "info 3.0.1 (core)",
    "limits 3.0.1 (core)",
    "login 2.0.1 (core)",
    "marketplace 1.0.2 (core)",
    "org 2.11.8 (core)",
    "schema 3.0.1 (core)",
    "settings 2.0.2 (core)",
    "sobject 0.2.15 (core)",
    "source 2.11.0 (core)",
    "telemetry 3.1.0 (core)",
    "templates 55.5.18 (core)",
    "trust 3.0.2 (core)",
    "user 2.3.42 (core)",
    "shane-sfdx-plugins 4.43.0 (user)"
  ]
}
```
### Additional information
Note: the output shell says 'cmd.exe' but I am actually running that from Powershell.

$PSVersionTable output:

```
Name                           Value
----                           -----
PSVersion                      5.1.19041.3031
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.3031
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

The doctor output of the command I shared in Windows:
```
[07:45:09.856] [90mTRACE[39m (sf:core:config): [36mCreated 'sf:core:config' child logger instance[39m
[07:45:09.856] [90mTRACE[39m (sf): [36mSetup child 'sf:core:config' logger instance[39m
[07:45:11.969] [90mTRACE[39m (sf:Config): [36mCreated 'sf:Config' child logger instance[39m
[07:45:11.969] [90mTRACE[39m (sf): [36mSetup child 'sf:Config' logger instance[39m
[07:45:11.970] [90mTRACE[39m (sf:Config): [36mCreated 'sf:Config' child logger instance[39m
[07:45:11.970] [90mTRACE[39m (sf): [36mSetup child 'sf:Config' logger instance[39m
[07:45:11.970] [32mINFO[39m (sf:Config): [36mReading config file: C:\Users\tarcan.gul\.sf\config.json[39m
[07:45:11.970] [32mINFO[39m (sf:Config): [36mReading config file: C:\Users\tarcan.gul\.commerce\devhubs\admin_tarcanATsdb3DOTcom\patch1_windowsATscratchDOTcom\.sf\config.json[39m
[07:45:12.000] [90mTRACE[39m (sf:OrgAccessor): [36mCreated 'sf:OrgAccessor' child logger instance[39m
[07:45:12.000] [90mTRACE[39m (sf): [36mSetup child 'sf:OrgAccessor' logger instance[39m
[07:45:12.000] [90mTRACE[39m (sf:SandboxAccessor): [36mCreated 'sf:SandboxAccessor' child logger instance[39m
[07:45:12.000] [90mTRACE[39m (sf): [36mSetup child 'sf:SandboxAccessor' logger instance[39m
[07:45:12.001] [90mTRACE[39m (sf:TokensConfig): [36mCreated 'sf:TokensConfig' child logger instance[39m
[07:45:12.001] [90mTRACE[39m (sf): [36mSetup child 'sf:TokensConfig' logger instance[39m
[07:45:12.001] [90mTRACE[39m (sf:crypto): [36mCreated 'sf:crypto' child logger instance[39m
[07:45:12.001] [90mTRACE[39m (sf): [36mSetup child 'sf:crypto' logger instance[39m
[07:45:12.001] [34mDEBUG[39m (sf:crypto): [36mretryStatus: undefined[39m
[07:45:12.001] [90mTRACE[39m (sf:keyChain): [36mCreated 'sf:keyChain' child logger instance[39m
[07:45:12.001] [90mTRACE[39m (sf): [36mSetup child 'sf:keyChain' logger instance[39m
[07:45:12.002] [34mDEBUG[39m (sf:keyChain): [36mplatform: win32[39m
[07:45:12.004] [32mINFO[39m (sf:TokensConfig): [36mReading config file: C:\Users\tarcan.gul\.sfdx\tokens.json[39m
[07:45:12.005] [90mTRACE[39m (sf:Org): [36mCreated 'sf:Org' child logger instance[39m
[07:45:12.005] [90mTRACE[39m (sf): [36mSetup child 'sf:Org' logger instance[39m
[07:45:12.005] [90mTRACE[39m (sf:AuthInfoConfig): [36mCreated 'sf:AuthInfoConfig' child logger instance[39m
[07:45:12.005] [90mTRACE[39m (sf): [36mSetup child 'sf:AuthInfoConfig' logger instance[39m
[07:45:12.006] [90mTRACE[39m (sf:crypto): [36mCreated 'sf:crypto' child logger instance[39m
[07:45:12.006] [90mTRACE[39m (sf): [36mSetup child 'sf:crypto' logger instance[39m
[07:45:12.006] [34mDEBUG[39m (sf:crypto): [36mretryStatus: undefined[39m
[07:45:12.006] [90mTRACE[39m (sf:keyChain): [36mCreated 'sf:keyChain' child logger instance[39m
[07:45:12.006] [90mTRACE[39m (sf): [36mSetup child 'sf:keyChain' logger instance[39m
[07:45:12.006] [34mDEBUG[39m (sf:keyChain): [36mplatform: win32[39m
[07:45:12.007] [32mINFO[39m (sf:AuthInfoConfig): [36mReading config file: C:\Users\tarcan.gul\.sfdx\patch1_windows@scratch.com.json[39m
[07:45:12.009] [90mTRACE[39m (sf:AuthInfo): [36mCreated 'sf:AuthInfo' child logger instance[39m
[07:45:12.009] [90mTRACE[39m (sf): [36mSetup child 'sf:AuthInfo' logger instance[39m
[07:45:12.015] [32mINFO[39m (sf:AuthInfo): [36mReturning fields for a connection using OAuth config.[39m
[07:45:12.017] [90mTRACE[39m (sf:connection): [36mCreated 'sf:connection' child logger instance[39m
[07:45:12.017] [90mTRACE[39m (sf): [36mSetup child 'sf:connection' logger instance[39m
[07:45:12.021] [34mDEBUG[39m (sf:connection): [36mAPI version cache last checked on 11/28/2023, 6:57:37 AM (now is 11/28/2023, 7:45:12 AM)[39m
[07:45:12.021] [34mDEBUG[39m (sf:connection): [36mUsing cached API version: 59.0[39m
[07:45:12.021] [34mDEBUG[39m (sf:connection): [36mConnection created with apiVersion 59.0[39m
[07:45:12.134] [90mTRACE[39m (sf:Org): [36mCreated 'sf:Org' child logger instance[39m
[07:45:12.134] [90mTRACE[39m (sf): [36mSetup child 'sf:Org' logger instance[39m
[07:45:12.134] [90mTRACE[39m (sf:AuthInfoConfig): [36mCreated 'sf:AuthInfoConfig' child logger instance[39m
[07:45:12.134] [90mTRACE[39m (sf): [36mSetup child 'sf:AuthInfoConfig' logger instance[39m
[07:45:12.135] [90mTRACE[39m (sf:crypto): [36mCreated 'sf:crypto' child logger instance[39m
[07:45:12.135] [90mTRACE[39m (sf): [36mSetup child 'sf:crypto' logger instance[39m
[07:45:12.135] [34mDEBUG[39m (sf:crypto): [36mretryStatus: undefined[39m
[07:45:12.135] [90mTRACE[39m (sf:keyChain): [36mCreated 'sf:keyChain' child logger instance[39m
[07:45:12.135] [90mTRACE[39m (sf): [36mSetup child 'sf:keyChain' logger instance[39m
[07:45:12.135] [34mDEBUG[39m (sf:keyChain): [36mplatform: win32[39m
[07:45:12.135] [32mINFO[39m (sf:AuthInfoConfig): [36mReading config file: C:\Users\tarcan.gul\.sfdx\patch1_windows@scratch.com.json[39m
[07:45:12.136] [90mTRACE[39m (sf:AuthInfo): [36mCreated 'sf:AuthInfo' child logger instance[39m
[07:45:12.136] [90mTRACE[39m (sf): [36mSetup child 'sf:AuthInfo' logger instance[39m
[07:45:12.140] [32mINFO[39m (sf:AuthInfo): [36mReturning fields for a connection using OAuth config.[39m
[07:45:12.140] [90mTRACE[39m (sf:connection): [36mCreated 'sf:connection' child logger instance[39m
[07:45:12.140] [90mTRACE[39m (sf): [36mSetup child 'sf:connection' logger instance[39m
[07:45:12.141] [34mDEBUG[39m (sf:connection): [36mAPI version cache last checked on 11/28/2023, 6:57:37 AM (now is 11/28/2023, 7:45:12 AM)[39m
[07:45:12.141] [34mDEBUG[39m (sf:connection): [36mUsing cached API version: 59.0[39m
[07:45:12.141] [34mDEBUG[39m (sf:connection): [36mConnection created with apiVersion 59.0[39m
[07:45:12.143] [90mTRACE[39m (sf:SfProjectJson): [36mCreated 'sf:SfProjectJson' child logger instance[39m
[07:45:12.143] [90mTRACE[39m (sf): [36mSetup child 'sf:SfProjectJson' logger instance[39m
[07:45:12.143] [32mINFO[39m (sf:SfProjectJson): [36mReading config file: C:\Users\tarcan.gul\.commerce\devhubs\admin_tarcanATsdb3DOTcom\patch1_windowsATscratchDOTcom\sfdx-project.json[39m
[07:45:12.148] [90mTRACE[39m (sf:Org): [36mCreated 'sf:Org' child logger instance[39m
[07:45:12.148] [90mTRACE[39m (sf): [36mSetup child 'sf:Org' logger instance[39m
[07:45:12.148] [90mTRACE[39m (sf:AuthInfoConfig): [36mCreated 'sf:AuthInfoConfig' child logger instance[39m
[07:45:12.148] [90mTRACE[39m (sf): [36mSetup child 'sf:AuthInfoConfig' logger instance[39m
[07:45:12.148] [90mTRACE[39m (sf:crypto): [36mCreated 'sf:crypto' child logger instance[39m
[07:45:12.148] [90mTRACE[39m (sf): [36mSetup child 'sf:crypto' logger instance[39m
[07:45:12.148] [34mDEBUG[39m (sf:crypto): [36mretryStatus: undefined[39m
[07:45:12.148] [90mTRACE[39m (sf:keyChain): [36mCreated 'sf:keyChain' child logger instance[39m
[07:45:12.148] [90mTRACE[39m (sf): [36mSetup child 'sf:keyChain' logger instance[39m
[07:45:12.148] [34mDEBUG[39m (sf:keyChain): [36mplatform: win32[39m
[07:45:12.149] [32mINFO[39m (sf:AuthInfoConfig): [36mReading config file: C:\Users\tarcan.gul\.sfdx\patch1_windows@scratch.com.json[39m
[07:45:12.149] [90mTRACE[39m (sf:AuthInfo): [36mCreated 'sf:AuthInfo' child logger instance[39m
[07:45:12.149] [90mTRACE[39m (sf): [36mSetup child 'sf:AuthInfo' logger instance[39m
[07:45:12.154] [32mINFO[39m (sf:AuthInfo): [36mReturning fields for a connection using OAuth config.[39m
[07:45:12.154] [90mTRACE[39m (sf:connection): [36mCreated 'sf:connection' child logger instance[39m
[07:45:12.154] [90mTRACE[39m (sf): [36mSetup child 'sf:connection' logger instance[39m
[07:45:12.154] [34mDEBUG[39m (sf:connection): [36mAPI version cache last checked on 11/28/2023, 6:57:37 AM (now is 11/28/2023, 7:45:12 AM)[39m
[07:45:12.154] [34mDEBUG[39m (sf:connection): [36mUsing cached API version: 59.0[39m
[07:45:12.154] [34mDEBUG[39m (sf:connection): [36mConnection created with apiVersion 59.0[39m
[07:45:12.154] [33mWARN[39m (sf:Org): [36mDefault API version is already 59.0[39m
[07:45:12.154] [90mTRACE[39m (sf:MetadataApiDeploy): [36mCreated 'sf:MetadataApiDeploy' child logger instance[39m
[07:45:12.154] [90mTRACE[39m (sf): [36mSetup child 'sf:MetadataApiDeploy' logger instance[39m
[07:45:12.156] [34mDEBUG[39m (sf:MetadataApiDeploy): [36mDeploying metadata source using SOAP v59.0[39m
[07:45:13.542] [34mDEBUG[39m (sf:MetadataApiDeploy): [36mStarted metadata transfer. ID = 0AfTC000002ODPi0AO[39m
[07:45:13.542] [90mTRACE[39m (sf:DeployCache): [36mCreated 'sf:DeployCache' child logger instance[39m
[07:45:13.542] [90mTRACE[39m (sf): [36mSetup child 'sf:DeployCache' logger instance[39m
[07:45:13.543] [32mINFO[39m (sf:DeployCache): [36mReading config file: C:\Users\tarcan.gul\.sf\deploy-cache.json[39m
[07:45:13.545] [32mINFO[39m (sf:DeployCache): [36mWriting to config file: C:\Users\tarcan.gul\.sf\deploy-cache.json[39m
[07:45:13.570] [90mTRACE[39m (sf:PollingClient): [36mCreated 'sf:PollingClient' child logger instance[39m
[07:45:13.570] [90mTRACE[39m (sf): [36mSetup child 'sf:PollingClient' logger instance[39m
[07:45:13.570] [34mDEBUG[39m (sf:MetadataApiDeploy): [36mPolling for metadata transfer status. ID = 0AfTC000002ODPi0AO[39m
[07:45:13.570] [34mDEBUG[39m (sf:MetadataApiDeploy): [36mPolling frequency (ms): 1000[39m
[07:45:13.570] [34mDEBUG[39m (sf:MetadataApiDeploy): [36mPolling timeout (min): 60[39m
[07:45:14.359] [34mDEBUG[39m (sf:MetadataApiDeploy): [36mMDAPI status update: Succeeded[39m
[07:45:14.362] [90mTRACE[39m (sf:DeployCache): [36mCreated 'sf:DeployCache' child logger instance[39m
[07:45:14.362] [90mTRACE[39m (sf): [36mSetup child 'sf:DeployCache' logger instance[39m
[07:45:14.362] [32mINFO[39m (sf:DeployCache): [36mReading config file: C:\Users\tarcan.gul\.sf\deploy-cache.json[39m
[07:45:14.363] [32mINFO[39m (sf:DeployCache): [36mWriting to config file: C:\Users\tarcan.gul\.sf\deploy-cache.json[39m
{
  "status": 0,
  "result": {
    "checkOnly": false,
    "completedDate": "2023-11-28T15:45:14.000Z",
    "createdBy": "005TC000000fG41",
    "createdByName": "User User",
    "createdDate": "2023-11-28T15:45:14.000Z",
    "details": {
      "componentSuccesses": [
        {
          "changed": true,
          "componentType": "",
          "created": false,
          "createdDate": "2023-11-28T15:45:14.000Z",
          "deleted": false,
          "fileName": "package.xml",
          "fullName": "package.xml",
          "success": true
        }
      ],
      "runTestResult": {
        "numFailures": 0,
        "numTestsRun": 0,
        "totalTime": 0,
        "codeCoverage": [],
        "codeCoverageWarnings": [],
        "failures": [],
        "flowCoverage": [],
        "flowCoverageWarnings": [],
        "successes": []
      },
      "componentFailures": []
    },
    "done": true,
    "id": "0AfTC000002ODPi0AO",
    "ignoreWarnings": true,
    "lastModifiedDate": "2023-11-28T15:45:14.000Z",
    "numberComponentErrors": 0,
    "numberComponentsDeployed": 0,
    "numberComponentsTotal": 0,
    "numberTestErrors": 0,
    "numberTestsCompleted": 0,
    "numberTestsTotal": 0,
    "rollbackOnError": true,
    "runTestsEnabled": false,
    "startDate": "2023-11-28T15:45:14.000Z",
    "status": "Succeeded",
    "success": true,
    "files": []
  },
  "warnings": []
}
```
I will not share the doctor command output in my macOS since it is long but I do get successes with the individual components.

**Comment by mshanemc** [comment]
huh.  I don't have my windows machine this week, sorry.  Can you try a few things for me?

[remove the = stuff for flags and refer to the --metadata-dir 
`sf project deploy start -o <your_scratch_org> -g --metadata-dir b2clocal21ToDeploy.zip -w 60 --verbose --api-version 59.0 --json --single-package`

[same but without --single-package]
`sf project deploy start -o <your_scratch_org> -g --metadata-dir b2clocal21ToDeploy.zip -w 60 --verbose --api-version 59.0 --json`

**Comment by tarcang** [comment]
> huh. I don't have my windows machine this week, sorry. Can you try a few things for me?
> 
> [remove the = stuff for flags and refer to the --metadata-dir `sf project deploy start -o <your_scratch_org> -g --metadata-dir b2clocal21ToDeploy.zip -w 60 --verbose --api-version 59.0 --json --single-package`
> 
> [same but without --single-package] `sf project deploy start -o <your_scratch_org> -g --metadata-dir b2clocal21ToDeploy.zip -w 60 --verbose --api-version 59.0 --json`

I will 👍

**Comment by tarcang** [comment]
Hello @mshanemc , I have tried the commands you have shared but unfortunately still got the same output. I have looked into this a little bit. Looking into the [deployment logic](https://github.com/forcedotcom/source-deploy-retrieve/blob/main/src/client/metadataApiDeploy.ts) I don't see any apparent reason why this wouldn't work other than the fact that there is something wrong with the `JSZip` library with its Windows compatibility. I am not sure, just putting out an idea on where it could be going wrong.

**Comment by tarcang** [comment]
Hello @mshanemc, any updates on this issue? I will try to help as much I can.

**Comment by mshanemc** [comment]
I've tried replicating this on windows.

I created a scratch org using this scratch definition file https://github.com/forcedotcom/commerce-on-lightning/blob/develop/config/b2b-project-scratch-def.json

When I deploy the zip file, I got back various errors (navigation menus that don't have network associated with them)

---

I also tried installing the commerce plugin and running `sf commerce scratchorg create` but that resulted in a lot of errors like `INVALID_EMAIL_ADDRESS`.  Looks like it requires a `--username` for some reason?  That at least got the org to create.

---

Then I deployed your zip file.  I got more failures.  

In all cases, I didn't have long deploys and it seemed to behave correctly (returning results for the errors).  

What am I missing?
